README.md

```diff
@@ -1,31 +1,348 @@
+
+```ruby
+require 'hyperion'
+
+route = RestRoute.new(:get, 'http://somesite.org/users/0', ResponseDescriptor.new('user', 1, :json))
+user = Hyperion.request(route)
+```
+
+You can pass `request` a block, in which case the return value of the
+block becomes the return value of `request`.
+
+```ruby
+user = Hyperion.request(route) do |result|
+  if result.status == HyperionResult::Status::SUCCESS
+    User.new(result.body)
+  end
+end
+```
+
+Production-quality error handling becomes hairy quickly, so hyperion
+provides a mini DSL to make it easier.
+
+```ruby
+Hyperion.request(route) do |result|
+  result.when(HyperionResult::Status::SUCCESS) { User.new(result.body) }
+  result.when(400..499) { raise 'we screwed up' }
+  result.when(500..599) { raise 'they screwed up' }
+  result.when(evil) { exit(1) }
+end
+
+def evil
+  proc do |result|
+    result.body['things'].any?{|x| x['id'] == '666'}
+  end
+end
+```
+
+Conditions are tested in order. When hyperion encounters the first
+true condition, it executes the associated block, the value of which
+becomes the return value of `request`.
+
+A condition may be:
+
+- a `HyperionResult::Status` enumeration member,
+- an `ErrorInfo::Code` enumeration member,
+- an HTTP code,
+- a range of HTTP codes, or
+- an arbitrary predicate.
+
+To obviate guard logic in predicates, a predicate that raises an
+exception is treated as a non-match. In the example above, if the body
+didn't have a `'things'` key, then `.any?` would raise a
+`NoMethodError`, interpreted as a non-match, and hyperion would move
+on to the next predicate.
+
+### Route classes
+
+In practice, you don't want to litter your code with `RestRoute.new`
+invocations. Here is a pattern for encapsulating the routes. It is the
+client-side analog of `routes.rb`.
+
+```ruby
+class CrudRoutes
+  def initialize(resource)
+    @resource = resource
+  end
+
+  def read(id)
+    build(:get, id, response(message_type_for(@resource), 1, :json))
+  end
+
+  def create
+    build(:post, '', response(message_type_for(@resource), 1, :json), payload(:json))
+  end
+
+  ...
+end
+```
+
+You can easily imagine the rest of the routes and what the helper
+methods `build`, `message_type_for`, `response`, and `payload` look like.
+
+Use it like this:
+
+```ruby
+user_routes = CrudRoutes.new('users')
+Hyperion.request(user_routes.create, body: {name: 'joe', email: 'joe@schmoe.com'})
+# later...
+joe = Hyperion.request(user_routes.read(joes_id))
+```
+
+A few notes:
+
+_TODO: comment on testing_
+
+_TODO: comment on DRYing_
+
+
+### Configuration
 
 ```ruby
-gem 'hyperion'
+# configure hyperion
+Hyperion.configure do |config|
+  config.vendor_string = 'indigobio-ascent'  # becomes part of the Accept header
+end
 ```
 
-And then execute:
 
-    $ bundle
+## Superion
+
+Superion layers more convenience onto Hyperion by helping dispatch the
+response: rendering the response as an entity and dealing with errors.
+
```

Comment:
@codyplungis In this document, I'm using it in the generally understood sense of the word. I did a little wordsmithing to make the first use of "entity" be in the phrase "an internal representation (often an entity)"